File: public/dsa/arrays/cumulative_sum/optimal_1/cumulative_sum.cpp
// imports-start
#include <bits/stdc++.h>
using namespace std;
// imports-end

vector<int> getCumulativeSum(vector<int> &arr) {
    int n = arr.size();
    vector<int> result(n);
    int prefixSum = 0;
    for (int i = 0; i < n; i++) {
        prefixSum += arr[i];
        result[i] = prefixSum;
    }
    return result;
}

// tests-start
int main() {
    vector<int> nums = {1, 2, 3, 4};
    auto result = getCumulativeSum(nums);
    for (int v : result) {
        cout << v << " ";
    }
    cout << endl;
    return 0;
}
// tests-end

/*output
1 3 6 10
*/

File: public/dsa/arrays/cumulative_sum/optimal_1/cumulative_sum.py

from typing import List


class Solution:
    def getCumulativeSum(self, arr: List[int]) -> List[int]:
        n = len(arr)
        result = [0 for _ in range(n)]
        prefix_sum = 0
        for i in range(n):
            prefix_sum += arr[i]
            result[i] = prefix_sum
        return result


# tests-start
arr = [1, 2, 3, 4]
sol = Solution()
print(sol.getCumulativeSum(arr))
# tests-end

'''output
[1, 3, 6, 10]
'''
